zkllvm/best-practices-limitations/rust-derive.mdx

```diff
@@ -0,0 +1,81 @@
+# Structs and enums in Rust
+
+Whenever a Rust circuit is compiled, `rustc` applies various optimizations to reduce its memory usage.
+
+Among these memory optimizations is [**reordering fields in structs and enums to avoid unnecessary 'paddings' in circuit IRs**](https://users.rust-lang.org/t/disable-struct-reordering-optimization-when-emitting-llvm-bitcode/74951). Consider the following example:
```

Comment:
I don't think it is a good idea to link to some random forum post in docs. If we really want to leave a link here, maybe it's better to link to Rust docs:

https://doc.rust-lang.org/reference/type-layout.html#representations

Or at to Nomicon:

https://doc.rust-lang.org/nomicon/repr-rust.html

Reply:
you are 100% right. I failed to find a proper explanation of this specific problem but now the link is fixed